code/script.js

```diff
@@ -221,105 +238,171 @@ const generateBoard = () => {
+          charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+        }
+      }
+      else if(category === 'hair' || category === 'eyes') {
+        if(keep) {
+          modal.style.display = "block"; // showing modal window to the user
+          document.getElementById('modal-text').innerHTML = `Yes, this person has ${value} ${category}` // changing text in the paragraph and showing user
+          charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+        } else {
+          modal.style.display = "block";
+          document.getElementById('modal-text').innerHTML = `No, this person hasn't ${value} ${category}`
+          charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+        }
+      }
+
+  // "close" (x) button that user can use to close the modal window
+  span.addEventListener('click', () => {
+    modal.style.display = "none";
+  })
 
+  // When the user clicks anywhere outside of the modal, close it
+  window.onclick = (event) => {
+    if (event.target == modal) {
+      modal.style.display = "none";
+    }
   }
+
 }
 
 // It'll filter the characters array and redraw the game board.
 const filterCharacters = (keep) => {
-  const { category, value } = currentQuestion
-  // Show the correct alert message for different categories
-  if (category === 'accessories') {
-    if (keep) {
-      alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
-      )
-    } else {
-      alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
-      )
-    }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  const { category, value } = currentQuestion // taking value from category and value
+  modalWindow(category, value, keep) // calling this function and passing 3 paramenters, this function filter and showing message to the user
```

Comment:
smart! :D

Reply:
Those functions can be a struggle, but when they work,  your code can be so much shorter and more easy to read :)